apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -75,6 +78,11 @@ const FormController = props => {
     warnOnExit = false
   } = props;
 
+  // [MEG] TODO: Once experiment is complete, use only allowPartialSaving prop
+  const usePartialSaving =
+    allowPartialSaving &&
+    experiments.isEnabled(experiments.TEACHER_APPLICATION_SAVING_REOPENING);
```

Comment:
Good call. That logic should happen in TeacherApplication anyways, not FormController. I went with using `useEffect`

Reply:
Hmm, now I'm thinking I can just put the `const` outside the component to avoid the renders. Still putting it in TeacherApplication

Reply:
> Hmm, now I'm thinking I can just put the const outside the component to avoid the renders. Still putting it in TeacherApplication

seems worth trying, but I would be worried that it will be hard to set/unset the experiment between different test runs in your unit tests.

Reply:
We don't have a testing file set up yet for TeacherApplication. I feel like the test would be "Is the experiment working?" since there's no logic now ... but I can put it in for documentation?

Reply:
I see what you are saying -- you can still test the different behaviors for FormController based on the prop you pass in, and for now we can just skip testing that TeacherApplication passes in a different prop based on the experiment. if that's all the experiment is doing then this approach seems ok to me. if we moved it up to new.js, there wouldn't be any test coverage of checking the experiment there anyway.

Reply:
Whoops, I missed this ... I moved it to new.js (I think it'll be easier to isolate there anyways)